src/arrayMethodSort.js

```diff
@@ -3,9 +3,36 @@
 /**
  * Implement method Sort
  */
+const charsCompare = (charA, charB) => {
+  if (('' + charA) > ('' + charB)) {
+    return 1;
+  }
+};
+
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(compareFunction = charsCompare) {
+    let counter = 0;
+
+    while (true) {
```

Comment:
Let's try not to use `while(true)`. It's never a good idea. Especially in such kind of tasks.
Let's try to change your loop. 
E.g. to 'do... while (counter > 0)' loop and annulate counter on the BEGINNING of each iteration and increase 'counter' in case of elements' swap.